delfin/drivers/ibm/ds8k/ds8k.py

```diff
@@ -0,0 +1,213 @@
+import six
+from oslo_log import log
+from oslo_utils import units
+
+from delfin import exception
+from delfin.common import constants
+from delfin.drivers import driver
+from delfin.drivers.ibm.ds8k import rest_handler, alert_handler
+
+LOG = log.getLogger(__name__)
+
+
+class DS8KDriver(driver.StorageDriver):
+
+    PORT_TYPE_MAP = {'FC-AL': constants.PortType.FC,
+                     'FICON': constants.PortType.FICON
+                     }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.rest_handler = rest_handler.RestHandler(**kwargs)
+        self.rest_handler.login()
+
+    def reset_connection(self, context, **kwargs):
+        self.rest_handler.logout()
+        self.rest_handler.verify = kwargs.get('verify', False)
+        self.rest_handler.login()
+
+    def close_connection(self):
+        self.rest_handler.logout()
+
+    def get_storage(self, context):
+        try:
+            system_info = self.rest_handler.get_rest_info('/api/v1/systems')
+            if system_info:
+                system_data = system_info.get('data', {}).get('systems')
+                for system in system_data:
+                    name = system.get('name')
+                    model = system.get('MTM')
+                    serial_number = system.get('sn')
+                    version = system.get('release')
+                    status = constants.StorageStatus.NORMAL
+                    if system.get('state') != 'online':
+                        status = constants.StorageStatus.ABNORMAL
+                    total = 0
+                    free = 0
+                    used = 0
+                    raw = 0
+                    if system.get('cap') != '' and \
+                            system.get('cap') is not None:
+                        total = int(system.get('cap'))
+                    if system.get('capraw') != '' and \
+                            system.get('capraw') is not None:
+                        raw = int(system.get('capraw'))
+                    if system.get('capalloc') != '' and \
+                            system.get('capalloc') is not None:
+                        used = int(system.get('capalloc'))
+                    if system.get('capavail') != '' and \
+                            system.get('capavail') is not None:
+                        free = int(system.get('capavail'))
+                    result = {
+                        'name': name,
+                        'vendor': 'IBM',
+                        'model': model,
+                        'status': status,
+                        'serial_number': serial_number,
+                        'firmware_version': version,
+                        'location': '',
+                        'total_capacity': total,
+                        'raw_capacity': raw,
+                        'used_capacity': used,
+                        'free_capacity': free
+                    }
+                    break
+            return result
```

Comment:
`result` might be referenced before assignment.

Reply:
done